unit_test/adapters/transforms/esdl_asset_mappers/test_esdl_asset_pipe_mapper.py

```diff
@@ -0,0 +1,138 @@
+#  Copyright (c) 2023. Deltares & TNO
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Test pipe mapper."""
+import unittest
+from pathlib import Path
+from unittest.mock import Mock
+
+from omotes_simulator_core.adapter.transforms.esdl_asset_mappers.pipe_mapper import (
+    EsdlAssetPipeMapper,
+)
+from omotes_simulator_core.entities.esdl_object import EsdlObject
+from omotes_simulator_core.infrastructure.utils import pyesdl_from_file
+from omotes_simulator_core.entities.assets.asset_defaults import PIPE_DEFAULTS
+
+
+class TestEsdlAssetPipeMapper(unittest.TestCase):
+    """Test class for pipe mapper."""
+
+    def setUp(self):
+        """Define the variables used in the tests."""
+        self.length: float = 1
+        self.inner_diameter: float = 1
+        self.roughness: float = 0.001
+        self.alpha_value: float = 0.8
+        self.minor_loss_coefficient: float = 0.0
+        self.external_temperature: float = 273.15 + 20.0
+        self.qheat_external: float = 0.0
+
+        # Arranging standard objects for the other test
+        esdl_file_path = (
+            Path(__file__).parent / ".." / ".." / ".." / ".." / "testdata" / "test1.esdl"
+        )
+        esdl_file_path = str(esdl_file_path)
+        self.esdl_object = EsdlObject(pyesdl_from_file(esdl_file_path))
+        self.mapper = EsdlAssetPipeMapper()
+
+    def test_to_entity_method(self):
+        """Test for to_entity method."""
+        pipes = self.esdl_object.get_all_assets_of_type("pipe")  # act
+        self.assertGreater(len(pipes), 0, "No pipes found in the ESDL file.")
+        self.esdl_asset = pipes[0]
+        esdl_asset = pipes[0]
+        pipe_entity = self.mapper.to_entity(esdl_asset)
+        self.assertEqual(pipe_entity.name, esdl_asset.esdl_asset.name)
+        self.assertEqual(pipe_entity.asset_id, esdl_asset.esdl_asset.id)
+        self.assertEqual(pipe_entity.connected_ports, esdl_asset.get_port_ids())  # type: ignore
+        self.assertEqual(
+            pipe_entity.length,
+            esdl_asset.get_property("length", PIPE_DEFAULTS.length)[0],  # type: ignore
+        )
+        self.assertEqual(
+            pipe_entity.diameter, self.mapper._get_diameter(esdl_asset)  # type: ignore
```

Comment:
I would suggest to use the same method for all calls: " esdl_asset.get_property("roughness", PIPE_DEFAULTS.roughness)[0]" or similar. Don't use the mapper methods in this specific case.

Reply:
Completed.